src/integrationtests/java/com/aws/greengrass/integrationtests/lifecyclemanager/ServiceDependencyLifecycleTest.java

```diff
@@ -181,8 +182,8 @@ void GIVEN_hard_dependency_WHEN_dependency_goes_through_lifecycle_events_THEN_cu
         Map<String, Object> configRemoveDep = new HashMap<String, Object>() {{
             put(SERVICES_NAMESPACE_TOPIC, new HashMap<String, Object>() {{
                 put("main", new HashMap<String, Object>() {{
-                    put(SERVICE_DEPENDENCIES_NAMESPACE_TOPIC, Arrays.asList(CustomerApp,
-                            DEFAULT_NUCLEUS_COMPONENT_NAME));
+                    put(SERVICE_DEPENDENCIES_NAMESPACE_TOPIC, new ArrayList<>(Arrays.asList(CustomerApp,
```

Comment:
It prevents the ClassCastException `class java.util.Arrays$ArrayList cannot be cast to class java.util.ArrayList`

Reply:
This is caused by the new validation code, a call is made to isDeviceConfiguredToTalkToCloud(), as a result the code will make a call to initializeNucleusComponentConfig() which tries to cast the object to (ArrayList)

https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java#L284

Reply:
After changing `ArrayList` to `List` in`initializeNucleusComponentConfig`, an **UnsupportedOperationException** occurs because `Arrays.asList` returns a fixed-sized list causing `mainDependencies.add` to fail

Reply:
Alright then, keep this change here. Don't change the cast.